ci/ci-lib.sh

```diff
@@ -2,28 +2,52 @@
 
 CI_DIR=$(dirname "$0")
 
+# Setup git
+git || (echo "FATAL: Failed to setup git!" && exit 1)
+git remote set-url origin git@github.com:sscarduzio/elasticsearch-readonlyrest-plugin.git
+git config --global push.default matching
+git config --global user.email "builds@travis-ci.com"
+git config --global user.name "Travis CI"
+
 # TAGGING
 function tag {
     GIT_TAG="$1"
 
-    # Check if this tag already exists, so we don't overwrite builds
-    if git tag --list | egrep -e "^${GIT_TAG}$" > /dev/null; then
-        echo "Git tag $GIT_TAG already exists, exiting."
-        return 1
-    fi
+    # Check if this tag already exists, so we don't overwrite existing builds
+    __tag_exists $GIT_TAG && return 1
 
     echo "Tagging as $GIT_TAG"
     if [[ "$(uname -s)" == *"Linux"* ]]; then
-        git remote set-url origin git@github.com:sscarduzio/elasticsearch-readonlyrest-plugin.git
-        git config --global push.default matching
-        git config --global user.email "builds@travis-ci.com"
-        git config --global user.name "Travis CI"
-        git tag $GIT_TAG -a -m "Generated tag from TravisCI build $TRAVIS_BUILD_NUMBER"
-        git push origin $GIT_TAG
+        git tag $GIT_TAG -a -m "Generated tag from TravisCI build $TRAVIS_BUILD_NUMBER" &&
+        git push origin $GIT_TAG || exit 1
+    fi
+    return 1
+}
+
+function __tag_exists {
+    GIT_TAG="$1"
+    if git pull && git tag --list | egrep -e "^${GIT_TAG}$" > /dev/null; then
+        return 0
+    fi
+    return 1
+}
+
+
+function tag_delete {
+    GIT_TAG="$1"
+    if __tag_exists; then
+      echo "No need to delete non-existing tag $GIT_TAG"
+      return 0
+    fi
+    echo "Deleting tag $GIT_TAG"
+    if [[ "$(uname -s)" == *"Linux"* ]]; then
+        git tag -d $GIT_TAG && git push origin :refs/tags/$GIT_TAG || (echo "Failed to delete tag $GIT_TAG" && exit 1)
+        return 0
     fi
     return 0
 }
 
+file . || (echo "FATAL: file command does not work or is not installed!" && exit 1)
```

Comment:
do we need it?

Reply:
This is checking a dependency for ci-uploader.sh, should belong to that file, indeed. 👍